<?xml version="1.0" ?>

<sdf version="1.6">

  <world name="ur5_robot_base">

    <gravity>0 0 -9.81</gravity>
    <physics name="default_physics" default="0" type="ode">
      <max_step_size>0.001</max_step_size>
      <real_time_factor>1</real_time_factor>
      <real_time_update_rate>1000</real_time_update_rate>
      <ode>                                            
        <solver>
          <type>quick</type>
          <iters>100</iters>
          <precon_iters>0</precon_iters>
          <sor>1.000000</sor>          
        </solver>
        <constraints>                                                         
          <cfm>0.000000</cfm>                                                 
          <erp>0.500000</erp>                                                 
          <contact_max_correcting_vel>1000.000000</contact_max_correcting_vel>
          <contact_surface_layer>0.00000</contact_surface_layer>               
        </constraints>
      </ode>          
    </physics>

    
    <scene>
      <ambient>0.4 0.4 0.4 1</ambient>
      <background>0.7 0.7 0.7 1</background>
      <shadows>1</shadows>
    </scene>

    <!-- Light Source -->
    <include>
      <uri>model://sun</uri>
    </include>

    <!-- A ground plane -->
    <include>
      <uri>model://ground_plane</uri>
      <pose>0 0 0 0 0 0</pose>
    </include>

    <!-- The robot base -->
    <model name="ur5_base">
      <include>
        <uri>model://ur5_base</uri>
      </include>
      <pose>0 0 0 0 0 0</pose>
    </model>


    <!-- Table -->
    <!-- Table top is at (x,y,z) = (0.7, 0, 0.775) -->
    <!-- Table top size is 0.9x0.9 -->
    <model name="cafe_table">
      <static>true</static>
      <include>
        <uri>model://cafe_table</uri>
      </include>
      <pose>0.7 0 -0.2 0 0 0</pose>
    </model>

    <model name="cafe_table_left">
      <include>
        <uri>model://ur5_base</uri>
      </include>
      <pose>0 0.55 0 0 0 0</pose>
    </model>
    <model name="cafe_table_right">
      <include>
        <uri>model://ur5_base</uri>
      </include>
      <pose>0 -0.55 0 0 0 0</pose>
    </model>
    
    
    <model name="storage_left">
      <static>true</static>
      <include>
        <uri>model://storage</uri>
      </include>
      <pose>0 0.55 0.6 0 0 0</pose>
    </model>

    <model name="storage_right">
      <static>true</static>
      <include>
        <uri>model://storage</uri>
      </include>
      <pose>0 -0.55 0.6 0 0 0</pose>
    </model>
    
  </world>
</sdf>
